cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/CredentialProfileStore.java

```diff
@@ -74,15 +111,21 @@ public Collection<CredentialProfileId> list(StructuredTableContext context, Stri
    * @throws IOException If any failure reading from storage occurs.
    */
   public Optional<CredentialProfile> get(StructuredTableContext context, CredentialProfileId id)
-      throws IOException {
+      throws CipherException, IOException {
     StructuredTable table = context.getTable(CredentialProviderStore.CREDENTIAL_PROFILES);
     Collection<Field<?>> key = Arrays.asList(
         Fields.stringField(CredentialProviderStore.NAMESPACE_FIELD,
             id.getNamespace()),
         Fields.stringField(CredentialProviderStore.PROFILE_NAME_FIELD,
             id.getName()));
-    return table.read(key).map(row -> GSON.fromJson(row
-        .getString(CredentialProviderStore.PROFILE_DATA_FIELD), CredentialProfile.class));
+    Optional<StructuredRow> row = table.read(key);
+    if (!row.isPresent()) {
```

Comment:
Similar to the comment above, it's better to use map instead of isPresent

Reply:
Done